OutOfSchool/OutOfSchool.WebApi/Services/ProviderServices/PublicProviderService.cs

```diff
@@ -45,6 +48,9 @@ public async Task<ProviderStatusDto> UpdateStatus(ProviderStatusDto dto, string
         // TODO: validate if current user has permission to update the provider status
         provider.Status = dto.Status;
         provider.StatusReason = dto.StatusReason;
+
+        changesLogService.AddEntityChangesToDbContext(provider, userId);
```

Comment:
(Logic) Not sure if it is the right place. You might get failure in next line (`providerRepository.UnitOfWork.CompleteAsync()`), but action is already logged

Reply:
`changesLogService.AddEntityChangesToDbContext(provider, userId)` not saves changes in db context, later we save changes by calling `CompleteAsync` on `UnitOfWork`
But added comment for clarity of how this works